pyopendds/DataReader.py

```diff
@@ -38,8 +37,14 @@ def take_next_sample(self) -> Any:
 
     def on_data_available_callback(self):
         sample = self.take_next_sample()
+        topicname = self.topic.name
+        print("on data available callback")
+        
         if sample is None:
             # print("on_data_available_callback error: sample is None")
             pass
         elif self.listener is not None:
-            self.listener(sample)
+            try: # if callback have 2 arguments
+                self.listener(sample,topicname)
```

Comment:
Can we have a callback with 2 arguments?

Reply:
I was unable to retrieve the topicname in the callback method (sub by function handler) . Maybe there are another solution ?